src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -81,6 +81,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Private ReadOnly _syntaxTrees As ImmutableArray(Of SyntaxTree)
 
         ''' <summary>
+        ''' The syntax trees explicitly given to the compilation at creation, in set form.
+        ''' </summary>
+        Private ReadOnly _syntaxTreeSet As HashSet(Of SyntaxTree)
```

Comment:
Is there an invariant that `IsEmbeddedOrMyTemplateTree` returns true for all compiler-generated (not user-provided) trees?